lnst/Tests/LongLivedConnections.py

```diff
@@ -0,0 +1,171 @@
+import time
+import select
+import socket
+import logging
+import resource
+import threading
+from typing import Union
+
+from .BaseTestModule import BaseTestModule
+from lnst.Common.IpAddress import Ip4Address, Ip6Address, ipaddress
+from lnst.Common.Parameters import IntParam, StrParam
+
+
+def get_af(ip: Union[Ip4Address, Ip6Address]):
+    ip = ipaddress(
+        ip
+    )  # self.params are transferred as values, so we need to convert it back
+
+    if isinstance(ip, Ip4Address):
+        return socket.AF_INET
+    elif isinstance(ip, Ip6Address):
+        return socket.AF_INET6
+    else:
+        raise ValueError(f"Unknown IP address type: {ip}")
+
+
+class BaseLongLivedTestModule(BaseTestModule):
+    server_ip = StrParam(mandatory=True)
+    server_port = IntParam(mandatory=True)
+    duration = IntParam(default=0)  # 0 means run indefinitely, until SIGINT is received
+    connections_count = IntParam(default=1)
+
+    def run(self):
+        self._running = True
+        soft, hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+
+        self._set_rlimit(hard, hard)
+        self._start()
+        if self.params.duration:
+            time.sleep(self.params.duration)
+        else:
+            try:
+                while True:
+                    time.sleep(1)
+            except KeyboardInterrupt:
+                logging.info("Interrupted, dying...")
+
+        self._stop()
+        self._set_rlimit(hard, soft)
+
+        return (
+            self._result
+        )  # is set as part of _stop (if # of conns matches expected # of conns)
+
+    def _set_rlimit(self, hard: int, soft: int):
```

Comment:
the order of these should be `(self, soft, hard)` so that we keep consistency with the `getrlimit` method...